### PR TITLE
Add test coverage for geolocation, and fix bug in the same

### PR DESCRIPTION
**Pull request by mythmon**
Bug 1262620

@Osmose r?

### CODE REVIEW

File: normandy/conftest.py [hunk 1]
```diff
@@ -22,3 +23,13 @@ def django_cache(request):
...
     for cache_name in settings.CACHES.keys():
         caches[cache_name].clear()
+
+
+@pytest.fixture
+def geolocation():
+    """Fixture to load geolocation data."""
+    geolocation_module.load_geoip_database()
+    if not geolocation_module.has_geoip:
+        pytest.skip()
```
Comment (mythmon) [inline normandy/conftest.py]: I want a way to make skips turn into failures in CI. Sumo has this with nose, and it helps us be sure about our CI. I can't find a thing in pytest to do this.

File: normandy/conftest.py [hunk 2]
```diff
@@ -22,3 +23,13 @@ def django_cache(request):
...
     for cache_name in settings.CACHES.keys():
         caches[cache_name].clear()
+
+
+@pytest.fixture
+def geolocation():
+    """Fixture to load geolocation data."""
+    geolocation_module.load_geoip_database()
+    if geolocation_module.geoip_reader is None:
+        pytest.skip()
```
Comment (mythmon) [inline normandy/conftest.py]: I want a way to make skips turn into failures in CI. Sumo has this with nose, and it helps us be sure about our CI. I can't find a thing in pytest to do this.

File: normandy/classifier/geolocation.py [hunk 3]
```diff
@@ -18,15 +18,18 @@ def load_geoip_database():
 
     try:
         geoip_reader = Reader(settings.GEOIP2_DATABASE)
-    except IOError:
+    except IOError as e:
```
Comment (willkg) [inline normandy/classifier/geolocation.py]: Nit: If you're not using the exception, you don't need the `as e`. Also, I'd suggest calling it `exc` or something that's not a single character.

Comment (mythmon) [inline normandy/classifier/geolocation.py]: Oops, this was a part of my debugging (I logged the error). I'll fix that.

**Comment by mythmon** [comment]
Ok. This is ready again.

**Comment by Osmose** [comment]
r+, nice work!

**Pull request merged**